src/SearchEngine.java

```diff
@@ -0,0 +1,111 @@
+import strategies.SearchStrategy;
+import strategies.StrategyALL;
+import strategies.StrategyANY;
+import strategies.StrategyNONE;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class SearchEngine {
+
+    Map<String, ArrayList<Integer>> map = new HashMap<>();
+    ArrayList<String> list = new ArrayList<>();
+    static Scanner scanner = new Scanner(System.in);
+
+    public void run(File file) {
+        ArrayList<String> peopleList = readFile(file);
+        Map<String, ArrayList<Integer>> words = mapWords(peopleList);
+        while (true) {
+            int selection = printMenu();
+            switch (selection) {
+                case 1 -> find(peopleList, words);
+                case 2 -> printList(peopleList);
+                case 0 -> {
+                    closeScanner();
+                    System.out.println("\nBye!");
+                    return;
+                }
+                default -> System.out.println("\nIncorrect option! Try again.");
+            }
+        }
+    }
+
+    public ArrayList<String> readFile(File file) {
+        try (Scanner fileReader = new Scanner(file)) {
+            while (fileReader.hasNextLine()) {
+                list.add(fileReader.nextLine());
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found.");
+        }
+        return list;
+    }
+
+    public Map<String, ArrayList<Integer>> mapWords(ArrayList<String> lines) {
+        int i = 0;
+        for (String s : lines) {
+            String[] words = s.split(" +");
+            for (String word : words) {
+                word = word.toUpperCase();
+                if (map.containsKey(word)) {
+                    map.get(word).add(i);
+                    map.put(word, map.get(word));
+                } else {
+                    ArrayList<Integer> arrayList = new ArrayList<>();
+                    arrayList.add(i);
+                    map.put(word, arrayList);
+                }
+            }
+            i++;
+        }
+        return map;
```

Comment:
Please rethink logic. 
Think about a class that keeps the information about `word` and `rowIndex`.

Reply:
Also, think about changing classic loops to stream API wherever possible.

Reply:
4 peoples helped me to fix this 🤣

Reply:
That's a bit better, but I still see the same problem. You want to keep the same arguments in the `search` method
`List<String> search(ArrayList<String> list, Map<String, ArrayList<Integer>> searchMap, String query);`
Probably it's not the best solution. 
What I mean here
https://github.com/Ksef/simple-search-engine/pull/1#discussion_r774113311
Use for example class
```
class LineTokenizer {
    private String line;
    private List<String> words;
    private int row;
}
```
or use another class that will contain the state of the line/words.

Reply:
seems to have succeeded